src/transformers/trainer.py

```diff
@@ -383,7 +383,14 @@ def setup_wandb(self):
             logger.info(
                 'Automatic Weights & Biases logging enabled, to disable set os.environ["WANDB_DISABLED"] = "true"'
             )
-            wandb.init(project=os.getenv("WANDB_PROJECT", "huggingface"), config=self.args.to_sanitized_dict())
+            combined_dict = {**self.model.config.to_dict(), **self.args.to_sanitized_dict()}
+            common_keys = self.model.config.to_dict().keys() & self.args.to_sanitized_dict().keys()
+            if common_keys:
+                warnings.warn(
```

Comment:
I wouldn't put a warning here as this will happen every time a user replaces a value in the config by using arguments (which is normal) and users get scared when they see a warning.